lib/vector-core/buffers/src/lib.rs

```diff
@@ -75,6 +80,12 @@ where
         } => {
             let (tx, rx) = mpsc::channel(max_events);
             let tx = BufferInputCloner::Memory(tx, when_full);
+            let rx = rx.inspect(|item| {
+                emit(&EventsSent {
+                    count: 1,
+                    byte_size: size_of_val(item),
```

Comment:
So here is where I think well want to use `allocated_bytes`, as mentioned in the above comment.